src/tokenizer.rs

```diff
@@ -620,44 +579,55 @@ macro_rules! advance_regex {
     };
 }
 
-fn advance_comment(code: &str) -> Advancement {
-    if let Some(beginning) = PATTERN_COMMENT_MULTI_LINE_BEGIN.find(code) {
-        if beginning.start() == 0 {
-            let block_count = beginning.end() - beginning.start() - "--[[".len();
-
-            let end_regex = Regex::new(&format!(r"\]={{{}}}\]", block_count)).unwrap();
+#[inline]
+fn parse_single_line_comment(code: &str) -> IResult<&str, &str> {
+    preceded(tag("--"), take_till(|x: char| x.is_ascii_control()))(code)
+}
 
-            let end_find = match end_regex.find(code) {
-                Some(find) => find,
-                None => return Err(TokenizerErrorType::UnclosedComment),
-            };
+#[inline]
+fn parse_multi_line_comment_start(code: &str) -> IResult<&str, &str> {
+    delimited(tag("--["), take_while(|x: char| x == '='), tag("["))(code)
+}
 
-            let comment = &code[beginning.end()..end_find.start()];
+#[inline]
+fn parse_multi_line_comment_body<'a>(
+    code: &'a str,
+    equality_symbol: &'a str,
```

Comment:
Not a great name, it's not checking the equality of anything, really. Last name was block_count, which isn't great but describes it better.